src/plugins/data_views/common/data_views/meta_units_to_formatter.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { FieldFormatParams } from '@kbn/field-formats-plugin/common';
+
+const timeUnitToDurationFmt = (inputFormat = 'milliseconds') => {
+  return {
+    id: 'duration',
+    params: {
+      inputFormat,
+      outputFormat: 'humanizePrecise',
+      outputPrecision: 2,
+      includeSpaceWithSuffix: true,
+      useShortSuffix: true,
+    },
+  };
+};
+
+export const metaUnitsToFormatter: Record<string, { id: string; params?: FieldFormatParams }> = {
+  percent: { id: 'percent' },
+  byte: { id: 'bytes' },
+  // need to consider if we want this, 1 microsecond will display as 0.001ms
+  micros: timeUnitToDurationFmt('microseconds'),
```

Comment:
I saw that you've removed the `nanos` formatter. I understand that there can be issues with the current formatting library. For example, 1 ns would be formatted as `0.00 ms`. However, I think that we should still support `nanos`, and file an issue for this known limitation/bug. In observability, we'd like to map the span duration as nanos. However, we don't expect spans to have a duration of only a few nanoseconds. It's much more typical to have something in the millisecond or second magnitude. So we'd miss not having support for `nanos`.

Reply:
So you want nanosecond values to display as millisecond even if nanosecond values less than 10,000 are displayed as 0.00 ms?

Reply:
I think that's better than not having a default formatter for nanoseconds by default, yes.

If this turns out to be a problem for a specific visualization, users can still change the formatting. This is just setting a default formatter if none has been configured explicitly. Did I get that right?

Reply:
Yes, thats correct. I'll restore the formatter